### PR TITLE
Update Envoy config to comply with 1.18.3 api

### PR DESCRIPTION
**Pull request by davinci26**
This patch makes the following updates:

1. Use `type.googleapis.com/envoy.extensions.access_loggers.stream.v3.StdoutAccessLog` in admin
2. Use `type.googleapis.com/envoy.extensions.access_loggers.stream.v3.StdoutAccessLog` in listeners
3. Use explicit http2 config

Signed-off-by: Sotiris Nanopoulos <sonanopo@microsoft.com>

### CODE REVIEW

File: pkg/envoy/xdsutil.go [hunk 1]
```diff
@@ -216,6 +217,26 @@ func GetUpstreamTLSContext(downstreamIdentity identity.ServiceIdentity, upstream
 	return tlsConfig
 }
 
+// GetHTTP2ProtocolOptions creates an Envoy http configuration that matches the downstream protocol
+func GetHTTP2ProtocolOptions() map[string]*any.Any {
```
Comment (shashankram) [inline pkg/envoy/xdsutil.go:221]: Would return an error and let the caller handle it instead of silently filling a nil value in a struct field.

**Review by shashankram — APPROVED** [review]

**Comment by davinci26** [comment]
@shashankram do you want to do another round of review

File: pkg/envoy/cds/cluster.go [hunk 2]
```diff
@@ -158,8 +166,12 @@ func getEgressClusters(clusterConfigs []*trafficpolicy.EgressClusterConfig) []*x
...
 			// Cluster config does not have a Host specified, route it to its original destination.
 			// Used for TCP based clusters
-			egressClusters = append(egressClusters, getOriginalDestinationEgressCluster(config.Name))
-
+			originalDestinationEgressCluster, err := getOriginalDestinationEgressCluster(config.Name)
+			if err != nil {
+				log.Error().Err(err).Msg("Error building the original destination cluster for the given egress cluster config")
+			} else {
+				egressClusters = append(egressClusters, originalDestinationEgressCluster)
+			}
```
Comment (shashankram) [inline pkg/envoy/cds/cluster.go:174]: nit: collapse conditional:
```suggestion
			if originalDestinationEgressCluster, err := getOriginalDestinationEgressCluster(config.Name);  err != nil {
				log.Error().Err(err).Msg("Error building the original destination cluster for the given egress cluster config")
			} else {
				egressClusters = append(egressClusters, originalDestinationEgressCluster)
			}
```

**Review by shashankram — APPROVED** [review]

File: pkg/envoy/cds/cluster.go (hunk 2 above)
Comment (davinci26) [reply pkg/envoy/cds/cluster.go]: neat golang feature

**Review by draychev — APPROVED** [review]

**Pull request merged**